src/runtime-rs/crates/resource/src/cgroups/utils.rs

```diff
@@ -14,3 +25,61 @@
 pub(crate) fn gen_overhead_path(path: &str) -> String {
     format!("kata_overhead/{}", path.trim_start_matches('/'))
 }
+
+pub(crate) fn rename_cgroup_path(path: &str) -> String {
+    if path.is_empty() || path == "/" {
+        return DEFAULT_RESOURCE_CONTROLLER_ID.to_string();
+    }
+
+    let cgroup_path = Path::new(path);
+    let cgroup_path_dir = cgroup_path.parent().unwrap().to_string_lossy().to_string();
+    let cgroup_path_name = format!(
+        "{}_{}",
+        CGROUP_KATA_PREFIX,
+        cgroup_path.file_name().unwrap().to_str().unwrap()
+    );
+
+    if cgroup_path_dir.is_empty() {
+        cgroup_path_name
+    } else {
+        format!("{}/{}", cgroup_path_dir, cgroup_path_name)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[derive(Debug)]
+    struct TestData {
+        path: String,
+        cgroup_path: String,
+    }
+
+    #[test]
+    fn test_rename_cgroup_path() {
+        let tests = &[
+            TestData {
+                path: "".to_string(),
+                cgroup_path: DEFAULT_RESOURCE_CONTROLLER_ID.to_string(),
+            },
+            TestData {
+                path: "/".to_string(),
+                cgroup_path: DEFAULT_RESOURCE_CONTROLLER_ID.to_string(),
+            },
+            TestData {
+                path: "hello".to_string(),
+                cgroup_path: "kata_hello".to_string(),
+            },
+            TestData {
+                path: "default/hello".to_string(),
+                cgroup_path: "default/kata_hello".to_string(),
+            },
+        ];
+
+        for (_i, t) in tests.iter().enumerate() {
```

Comment:
It is not necessary to have an `enumerate()`.

Reply:
deleted

Reply:
After moved the `enumerate()`, then the for loop should be `for t in tests.iter()` instead of `for (_i, t) in tests.iter()`.

Reply:
fixed